packages/forms/src/styled/radio/StyledRadioInput.ts

```diff
@@ -135,6 +140,7 @@ export const StyledRadioInput = styled.input.attrs({
       box-shadow .1s ease-in-out,
       background-color .25s ease-in-out,
       color .25s ease-in-out;
+    z-index: -1;
```

Comment:
Without a [stacking context](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context) reset, these visual artifacts are susceptible to being hidden by an intervening stacked layer (i.e. absolute `div`). I think there are a couple of things that could be done here:
- Add `z-index: 0;` to the `StyledRadioLabel`
-or-
- Update the `Radio`, `Checkbox`, `Toggle` element components to render the styled input _after_ the `{children}` and remove the `z-index: -1;` here

I'd generally be in favor of the latter over trying to manipulate stacking contexts – which always feels a bit messy.  But only if it can be done with minimal CSS updates. If we find ourselves updating a bunch of CSS to make it work, then I'd advise sticking with the former.

The problem can be seen with Bedrock enabled in Storybook: https://61e9d60fd1320011fdcba22c--zendeskgarden-react-components.netlify.app/?path=/story/packages-forms-fieldset--checkbox&globals=bedrock:enabled

Reply:
Actually, the more I think about it, I think there are a bunch of sibling selectors that break if the `{children}`/input are reordered. So I'd recommend sticking with the `z-index: 0;` added to `StyledRadioLabel`.

Reply:
@jzempel — thanks for showing me the Bedrock thing! (I didn't know about it before.)

> I'd generally be in favor of the latter over trying to manipulate stacking contexts – which always feels a bit messy.
I totally agree with you here. In this particular case, I think part of the challenge comes from the faux input (the `label::before`) being positioned absolutely, as well as the `input`. I can't help but wonder if styling whole component with Flexbox instead might help alleviate this. 🤔 

RE the `z-index`: I just tried it out again locally — both with and without Bedrock — and it looks like we might not need to hard-set the `z-index` after all? I just pushed up a commit that removes it altogether. Would you mind taking another look?